File: src/main/java/burst/pool/miners/Payable.java
package burst.pool.miners;

import burst.kit.entity.BurstAddress;
import burst.kit.entity.BurstValue;

public interface Payable {
    void increasePending(BurstValue delta);
    void decreasePending(BurstValue delta);
    BurstValue getMinimumPayout();
    BurstValue takeShare(BurstValue availableReward);
    BurstValue getPending();
    BurstAddress getAddress();
}

File: src/main/kotlin/burst/pool/miners/Payable.kt
package burst.pool.miners

import burst.kit.entity.BurstAddress
import burst.kit.entity.BurstValue

interface Payable {
    fun increasePending(delta: BurstValue)
    fun decreasePending(delta: BurstValue)
    val minimumPayout: BurstValue
    fun takeShare(availableReward: BurstValue): BurstValue
    val pending: BurstValue
    val address: BurstAddress
}
